grpc/observability/config.proto

```diff
@@ -0,0 +1,142 @@
+
+    // NeverSampler disables sampling.
+    message NeverSampler {}
+
+    oneof sampler {
+        RateLimitingSampler rate_limiting = 1;
+        ProbabilisticSampler probabilistic = 2;
+        AlwaysSampler always = 3;
+        NeverSampler never = 4;
+    }
+}
+
+message ExporterConfig {
+    // Following options turn off the default exporters, this allows
+    // applications to use custom exporters without duplicated uploads.
+    bool disable_default_tracing_exporter = 1;
+    bool disable_default_metrics_exporter = 2;
+    bool disable_default_logging_exporter = 3;
+
+    // The project identifier.
+    string project_id = 4;
+
+    // The reporting interval for all observability data for default exporters.
+    google.protobuf.Duration reporting_interval = 5;
+}
+
+message TracingConfig {
+    // Defines the sampling algorithm and behavior. The default sampler is
+    // RateLimitingSampler with 10s interval.
+    Sampler sampler = 1;
+}
+
+message MetricsConfig {
+}
+
+message LoggingConfig {
+    message LogFilter {
+        // A string which can select a group of method names. Only "*" wildcard
+        // is accepted.
+        // Examples:
+        // - "Foo/Bar" selects only the method "Bar" from service "Foo"
+        // - "Foo/*" selects all methods from service "Foo"
+        // - "*/*" selects all methods from all services.
+        string pattern = 1;
+        // Number of bytes of each header to log. If the size of the header is
+        // greater than the defined limit, content pass the limit will be
+        // truncated. The default value is 0.
+        int32 header_bytes = 2;
+        // Number of bytes of each message to log. If the size of the message is
+        // greater than the defined limit, content pass the limit will be
+        // truncated. The default value is 0.
+        int32 message_bytes = 3;
+        // Attachs a sampler to this group of methods. The default sampler is
+        // AlwaysSampler.
+        Sampler sampler = 4;
+    }
+
+    // A list of method config. The order matters here - the first pattern which
+    // matches the current method will apply the associated config options in
+    // the LogFilter. Any other LogFilter that also matches that comes later
+    // will be ignored. So a LogFilter of "*/*" should appear last in this list.
+    repeated LogFilter log_filters = 1;
+
+    // If specified, only this list of events will be logged. If the list is
+    // empty, all the events will be logged.
+    repeated grpc.observability.logging.v1.GrpcLogRecord.EventType event_types = 2;
+}
+
+message DeploymentConfig {
+}
+
+// Configuration for observability behaviors. By default, no configuration is
+// required for tracing/metrics/logging to function. This config captures the
+// most common knobs for gRPC users. It's always possible to override with
+// explicit config in code.
+message ObservabilityConfig {
+    // Controls the exporter behavior
+    ExporterConfig exporter = 1;
```

Comment:
`exporter_config` (etc)?

Should these message types be nested inside `ObservabilityConfig`?

Reply:
Done and done. I left `Sampler` out, since it might be directly referenced else where.